### PR TITLE
Refactor reference updaters

### PR DESCRIPTION
**Pull request by illusional**
- Remove old update_reference_server in favour of new cloud run jobs - these have a longer timeout and just run a command from the container directly.
- Introduced a new _apply schema_ cloud run job that gets run just after deploy automatically.
- Adds a script that describes how the cloud run jobs were created
- Add missing migration.

I want to include the commands I used to deploy the cloud run services, but they do contain paths and IP addresses, it looks something like this:

```bash
gcloud beta run jobs create \
   staging-reference-update-hpo-terms \
   [... other config options]
```

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -59,6 +59,17 @@ jobs:
       - name: push image
         run: docker push --all-tags $DOCKER_IMAGE
 
+      - name: Apply new schema
+        if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/staging'
+        run: |
+          # cloud run jobs to update
+          run_job_names=("$DEPLOY_TARGET-schema-migrator" "$DEPLOY_TARGET-reference-schema-migrator" "$DEPLOY_TARGET-reference-updater" "$DEPLOY_TARGET-reference-update-hpo-terms" "$DEPLOY_TARGET-reference-import-panels")
+          for cloud_run_target in run_job_names; do
```
Comment (lgruen) [inline .github/workflows/deploy.yaml:67]: Just double-checking that it's okay to run these updates concurrently? I.e. there will never be any interdependencies or tables that are written to by the same jobs?

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -59,6 +59,17 @@ jobs:
...
+          
+            gcloud beta run jobs update \
```
Comment (lgruen) [inline .github/workflows/deploy.yaml:69]: Should we wait on these jobs to finish successfully within the CI?

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -67,11 +78,10 @@ jobs:
...
-      - name: restart seqr reference-updater
-        if: github.ref == 'refs/heads/main'
+      - name: apply schema
         run: |
-          # Used for Cloud Scheduler tasks.
-          gcloud run deploy \
-            seqr-$DEPLOY_TARGET-reference-updater \
-            --region australia-southeast1 \
-            --image=$DOCKER_IMAGE:gcloud-$DEPLOY_TARGET
+          gcloud beta run jobs execute \
```
Comment (lgruen) [inline .github/workflows/deploy.yaml:83]: Ditto.

**Review by lgruen — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yaml (hunk 2 above)
Comment (illusional) [reply .github/workflows/deploy.yaml:69]: Yep!

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (illusional) [reply .github/workflows/deploy.yaml:67]: These don't get run at the same time, this just updates the image dependency. I've split out the auto reference updaters, and schema is applied separately (because it's got the `--wait` on it).

Comment (lgruen) [reply .github/workflows/deploy.yaml:67]: Thanks very much for explaining!

File: deploy/docker/seqr/deploy_cloudrun_jobs.sh [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+POSTGRES_USERNAME="postgres"
+POSTGRES_DATABASE="seqrdb"
+POSTGRES_REFERENCE_DATABASE="reference_data_db"
+POSTGRES_PASSWORD_SECRET_NAME="postgres-password"
+
+GSA_SECRET_NAME="gsa-key-prod"
+DJANGO_SECRET_NAME="django-key-prod"
+
+gcloud beta run jobs create \
```
Comment (lgruen) [inline deploy/docker/seqr/deploy_cloudrun_jobs.sh:17]: Not super important, but given that only the `args` seem to differ below, a `for` loop could avoid a lot of the code duplication.

File: deploy/docker/seqr/deploy_cloudrun_jobs.sh [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+   --region=australia-southeast1 \
+   --service-account=${SERVICE_ACCOUNT} \
+   --vpc-connector=projects/seqr-308602/locations/australia-southeast1/connectors/seqr-cloud-run-to-sql \
+   --vpc-egress=private-ranges-only \
+   --set-env-vars="DEPLOYMENT_TYPE=dev,BASE_URL=${SEQR_URL},POSTGRES_SERVICE_HOSTNAME=10.94.145.3,POSTGRES_SERVICE_PORT=5432,POSTGRES_DATABASE=${POSTGRES_DATABASE},POSTGRES_REFERENCE_DATABASE=${POSTGRES_REFERENCE_DATABASE},POSTGRES_USERNAME=${POSTGRES_USERNAME},PYTHONPATH=/seqr,PYTHONUNBUFFERED=True" \
+   --set-secrets="DJANGO_KEY=${DJANGO_SECRET_NAME}:latest,GSA_KEY=${GSA_SECRET_NAME}:latest,POSTGRES_PASSWORD=${POSTGRES_PASSWORD_SECRET_NAME}:latest" \
+   --execute-now
+
+
+# Add triggers
```
Comment (lgruen) [inline deploy/docker/seqr/deploy_cloudrun_jobs.sh:83]: Might be worth mentioning that (and why) these are spaced out time-wise.

File: seqr/migrations/0047_alter_project_is_mme_enabled.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.15 on 2022-10-14 03:07
```
Comment (lgruen) [inline seqr/migrations/0047_alter_project_is_mme_enabled.py:1]: Did you include this deliberately? How is this related to the PR?

**Review by lgruen — APPROVED** [review]
Thanks! I've added some minor comments. Please don't forget to update the PR description.

File: seqr/migrations/0047_alter_project_is_mme_enabled.py (hunk 6 above)
Comment (illusional) [reply seqr/migrations/0047_alter_project_is_mme_enabled.py:1]: Ah yes I did, this should have been included in the merge - I checked and the schema appliers should not be makingmigrations, these should be included and they should just be applied.

File: deploy/docker/seqr/deploy_cloudrun_jobs.sh (hunk 4 above)
Comment (illusional) [reply deploy/docker/seqr/deploy_cloudrun_jobs.sh:17]: I thought about it, the main difference is the reference updater uses more memory - not planning to use it a lot (more for reference) so for now I'll just leave it.
